flink-end-to-end-tests/flink-end-to-end-tests-common/src/main/java/org/apache/flink/tests/util/FlinkDistribution.java

```diff
@@ -0,0 +1,219 @@
+import org.apache.flink.configuration.UnmodifiableConfiguration;
+import org.apache.flink.util.ExceptionUtils;
+
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.JsonNode;
+import org.apache.flink.shaded.jackson2.com.fasterxml.jackson.databind.ObjectMapper;
+
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.Response;
+import org.junit.Assert;
+import org.junit.rules.ExternalResource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.StandardCopyOption;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+/**
+ * A wrapper around a Flink distribution.
+ */
+public final class FlinkDistribution extends ExternalResource {
+
+	private static final Logger LOG = LoggerFactory.getLogger(FlinkDistribution.class);
+
+	private static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+	private final List<AutoClosablePath> filesToDelete = new ArrayList<>(4);
+
+	private static final Path FLINK_CONF_YAML = Paths.get("flink-conf.yaml");
+	private static final Path FLINK_CONF_YAML_BACKUP = Paths.get("flink-conf.yaml.bak");
+
+	private final Path opt;
+	private final Path lib;
+	private final Path conf;
+	private final Path log;
+	private final Path bin;
+
+	private Configuration defaultConfig;
+
+	public FlinkDistribution() {
+		final String distDirProperty = System.getProperty("distDir");
+		if (distDirProperty == null) {
+			Assert.fail("The distDir property was not set. You can set it when running maven via -DdistDir=<path> .");
+		}
+		final Path flinkDir = Paths.get(distDirProperty);
+		bin = flinkDir.resolve("bin");
+		opt = flinkDir.resolve("opt");
+		lib = flinkDir.resolve("lib");
+		conf = flinkDir.resolve("conf");
+		log = flinkDir.resolve("log");
+	}
+
+	@Override
+	protected void before() throws IOException {
+		defaultConfig = new UnmodifiableConfiguration(GlobalConfiguration.loadConfiguration(conf.toAbsolutePath().toString()));
+		final Path originalConfig = conf.resolve(FLINK_CONF_YAML);
+		final Path backupConfig = conf.resolve(FLINK_CONF_YAML_BACKUP);
+		Files.copy(originalConfig, backupConfig);
+		filesToDelete.add(new AutoClosablePath(backupConfig));
+	}
+
+	@Override
+	protected void after() {
+		try {
+			stopFlinkCluster();
+		} catch (IOException e) {
+			LOG.error("Failure while shutting down Flink cluster.", e);
+		}
+
+		final Path originalConfig = conf.resolve(FLINK_CONF_YAML);
```

Comment:
As before I would create the method `restoreDefaultConfig` and also call it here.